src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -1054,22 +1054,22 @@ private void AddComponentStats(StringBuilder statsBuilder)
                 statsBuilder.AppendLine($"Queueued downloads: {unassignedDownloads}");
             }
 
-            double averageBlockSizeKb = this.GetAverageBlockSizeBytes() / 1024.0;
+            double avgBlockSizeBytes = this.GetAverageBlockSizeBytes();
+            double averageBlockSizeKb = avgBlockSizeBytes / 1024.0;
             statsBuilder.AppendLine($"Average block size: {Math.Round(averageBlockSizeKb, 2)} KB");
 
-            double totalSpeedKB = (this.GetTotalSpeedOfAllPeersBytesPerSec() / 1024.0);
-            statsBuilder.AppendLine($"Total download speed: {Math.Round(totalSpeedKB, 2)} KB/sec");
-
-            // TODO: do that when component is activated.
-            // TODO move to nodestats and not bench
-            // just for logging
-            // show it as a part of nodestats, not separated spammer:
-            // avg download speed
-            // peer quality score (sort by quality score)
-            // number of assigned blocks
-            // MaxBlocksBeingDownloaded
-            // amount of blocks being downloaded
-            // show actual speed (no 1mb limit)
+            double totalSpeedBytesPerSec = this.GetTotalSpeedOfAllPeersBytesPerSec();
+            double totalSpeedKbPerSec = (totalSpeedBytesPerSec / 1024.0);
+            statsBuilder.AppendLine($"Total download speed: {Math.Round(totalSpeedKbPerSec, 2)} KB/sec");
+
+            double timeToDownloadBlockMs = Math.Round((avgBlockSizeBytes / totalSpeedBytesPerSec) * 1000, 2);
```

Comment:
This might propagate all the way down here?

Reply:
Not sure what you've meant by that. Cal you elaborate?

Reply:
I meant that if something higher up was 0 then it be an issue with the calculations but you explained it so all good 👍